local_planner/src/nodes/waypoint_generator.cpp

```diff
@@ -208,40 +208,31 @@ void WaypointGenerator::reachGoalAltitudeFirst() {
   pose_to_wp *= planner_info_.min_speed;
 
   output_.goto_position = toPoint(toEigen(pose_.pose.position) + pose_to_wp);
+  output_.adapted_goto_position = output_.goto_position;
+  output_.smoothed_goto_position = output_.goto_position;
+  smoothed_goto_location_ = toEigen(output_.smoothed_goto_position);
 }
 
 void WaypointGenerator::smoothWaypoint(double dt) {
-  Eigen::Vector2f vel_cur_xy(curr_vel_.twist.linear.x,
-                             curr_vel_.twist.linear.y);
-  Eigen::Vector2f vel_sp_xy((output_.adapted_goto_position.x -
-                             last_position_waypoint_.pose.position.x) /
-                                dt,
-                            (output_.adapted_goto_position.y -
-                             last_position_waypoint_.pose.position.y) /
-                                dt);
-
-  Eigen::Vector2f accel_diff = (vel_sp_xy - vel_cur_xy) / dt;
-  Eigen::Vector2f accel_cur = (vel_cur_xy - last_velocity_) / dt;
-  Eigen::Vector2f jerk_diff = (accel_diff - accel_cur) / dt;
-  float max_jerk = max_jerk_limit_param_;
-
-  // velocity-dependent max jerk
-  if (min_jerk_limit_param_ > 0.001f) {
-    max_jerk *= vel_cur_xy.norm();
-    if (max_jerk < min_jerk_limit_param_) max_jerk = min_jerk_limit_param_;
-  }
-
-  if (jerk_diff.squaredNorm() > max_jerk * max_jerk && max_jerk > 0.001f) {
-    jerk_diff = max_jerk * jerk_diff.normalized();
-    vel_sp_xy = (jerk_diff * dt + accel_cur) * dt + vel_cur_xy;
+  if (smoothing_speed_ > 0.01) {
+    const float P_constant = smoothing_speed_;
+    const float D_constant = 2.f * std::sqrt(P_constant);  // critically damped
+    const Eigen::Vector3f desired_location =
+        toEigen(output_.adapted_goto_position);
+    const Eigen::Vector3f desired_velocity = toEigen(curr_vel_.twist.linear);
```

Comment:
This is the desired velocity of the setpoint, not of the drone. The idea is that if the drone is moving, the setpoint should move too, since it should stay a constant distance in front of the drone. So we don't penalize movement of the setpoint in general, just movement relative to the drone.